code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -465,45 +373,34 @@
 
 	data["production_mode"] = production_mode
 
-	data["pillname"] = pillname
-	data["patchname"] = patchname
-	data["bottlename"] = bottlename
-
-	data["maxpills"] = CHEMMASTER_MAX_PILLS
-	data["maxpatches"] = CHEMMASTER_MAX_PATCHES
-	data["maxbottles"] = CHEMMASTER_MAX_BOTTLES
-
-	if(reagents.total_volume)
-		var/amount_per_pill = clamp(reagents.total_volume / pillamount, 0, MAX_UNITS_PER_PILL)
-		data["pillplaceholdername"] = "[reagents.get_master_reagent_name()] ([amount_per_pill]u)"
-		var/amount_per_patch = clamp(reagents.total_volume / patchamount, 0, MAX_UNITS_PER_PATCH)
-		data["patchplaceholdername"] = "[reagents.get_master_reagent_name()] ([amount_per_patch]u)"
-		data["bottleplaceholdername"] = reagents.get_master_reagent_name()
-
 	return data
 
 /obj/machinery/chem_master/ui_static_data(mob/user)
 	var/list/data = list()
 
 	data["maxnamelength"] = MAX_CUSTOM_NAME_LEN
 
-	var/pill_styles = list()
-	for(var/i in 1 to MAX_PILL_SPRITE)
-		pill_styles += list(list(
-			"id" = i,
-			"sprite" = "pill[i]",
-		))
-	data["pillstyles"] = pill_styles
-
-	var/bottle_styles_with_sprite = list()
-	var/bottle_style_indexer = 0
-	for(var/style in bottle_styles)
-		bottle_style_indexer++
-		bottle_styles_with_sprite += list(list(
-			"id" = bottle_style_indexer,
-			"sprite" = "[style]",
-		))
-	data["bottlestyles"] = bottle_styles_with_sprite
+	var/static_production_data = list()
+	var/production_data_indexer = 0
+	for(var/datum/chemical_production_mode/M in production_modes)
+		var/mode_data = list(
+			"id" = ++production_data_indexer,
```

Comment:
It would also mean we can change this to be something like this instead of indexes
```suggestion
			"id" = M.mode_name,
```